### DISCUSSION TITLE
This is NUTS! 7M+ reqs/s

### DISCUSSION THREAD

**Discussion by deluxor** [kartikk221/hyper-express#130] [Show and tell]
Guys this is just nuts!

I just achieved a forked hello world REST call doing 7 million requests per second on my machine, running linux subsystem for windows!

[https://asciinema.org/a/iIxiZrZqetcYdTk473igq1jFB](https://asciinema.org/a/iIxiZrZqetcYdTk473igq1jFB)

What a wonderful thing!

**Comment by piliugin-anton** [comment]
Try to benchmark with wrk (https://github.com/wg/wrk) you will get more realistic results

**Comment by kartikk221** [comment]
Yep, as @piliugin-anton use a different benchmarking utility then autocannon, I'll be re-doing the benchmark section of the repo soon with a reliable benchmark utility instead of autocannon.

**Comment by deluxor** [comment]
Well, using WRK it is way less reqs/s, but still really good ~1.4M req/s:
```
root@Desktop:~# wrk -c 500 -d 30 -t 30 http://127.0.0.1:8080/clients
Running 30s test @ http://127.0.0.1:8080/clients
  30 threads and 500 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency   274.47us    1.04ms  64.26ms   98.83%
    Req/Sec    47.56k     5.80k  100.65k    71.72%
  42695190 requests in 30.10s, 4.02GB read
Requests/sec: 1418506.68
Transfer/sec:    136.63MB
root@Desktop:~#
```
But just keep in mind that I'm running the server and the benchmark util on the same machine, and it is running on a virtualized environment.

**Comment by isaadabbasi** [comment]
On my m1x machine:MacOS catalina
I tried benchmarking it with ab and autocannon. The server doesn’t seem to reply to even 1 request.

**Reply by kartikk221** [reply]
Hey, you would have to provide a reproducible example. It sounds like the URL you are benchmarking is not serving as intended due to a setup error on your end.
